### PR TITLE
[ONNX] Extend ONNX Frontend with Function Mish-18

### LINKED ISSUE #20550: [Good First Issue]: Extend ONNX Frontend with Function Mish-18
### Context

Neural networks are graphs consisting of nodes called operators. Each operator corresponds to a mathematical function, usually described in framework's documentation or an AI standard, such as ONNX.
OpenVINO ONNX Frontend is a component responsible for working with ONNX graphs and requires implementation of different ONNX operators in order to use ONNX models.
This task requires extending OpenVINO ONNX Frontend with Function Mish.
Necessary help will be provided by ONNX Fronted team.

### What needs to be done?

Operator details can be found in [ONNX Operators](https://github.com/onnx/onnx/blob/main/docs/Operators.md#Mish)
More details can be found in [ONNX Changelog](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#Mish-18)
1. Create `.hpp` and `.cpp` files [here](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/frontend/src/op)
2. Prepare an implementation of this operator in form of a function. Mish-18 should be placed in [opset 1 namespace](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/op/unique.hpp#L13).
3. Register the function in [ops_bridge.cpp](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/ops_bridge.cpp#L554-L560) while keeping alphabetical order
4. Create test model(s) in [ONNX models directory](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/tests/models/). OpenVINO test infrastructure then converts prototxt files to ONNX models - you will use those models later in tests
5. Add tests covering all use cases [here](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/onnx_import.in.cpp)
6. Check [Python xfailed tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/tests_python/test_backend.py) to find a test marked as a xfailed for added functionality. If any exist - remove corresponding lines and try to verify by using cmdline "python -m pytest -k name_of_test".
More details in [adding operators to ONNX Frontend guide](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/how_to_add_op.md)

### Example Pull Requests

- https://github.com/openvinotoolkit/openvino/pull/15076
- https://github.com/openvinotoolkit/openvino/pull/14936
- https://github.com/openvinotoolkit/openvino/pull/12510
- https://github.com/openvinotoolkit/openvino/pull/19428

### Resources

- [What is OpenVINO?](https://github.com/openvinotoolkit/openvino#what-is-openvino-toolkit)
- [Contribution guide](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [Blog post on contributing to OpenVINO](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [User documentation](https://docs.openvino.ai/)
- Guides on [ONNX Frontend architecture](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/architecture.md) and [ONNX Frontend tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/tests.md) are also available
- You can also see a [blog post on contributing to OpenVINO](https://medium.com/openvino-toolkit/how-to-contribute-to-an-ai-open-source-project-c741f48e009e)

### Contact points

@gkrivor

### Ticket

_No response_

### PR DESCRIPTION
**Pull request by sydarb**
### Details:
 - Implemented ONNX frontend Mish-18 operator and corresponding tests.

### Tickets:
 - Closes #20550

### CODE REVIEW

**Comment by sydarb** [comment]
Hi @gkrivor, can you guide me on how to replicate the test which is failing, on my local machine (intel macos). So far I am able to test with c++ gtest (ov_onnx_frontend_tests) and all were passing. However I was not able to follow the steps mentioned for [python frontend tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/tests.md#python-frontend-tests).

**Comment by p-wysocki** [comment]
build_jenkins

**Comment by gkrivor** [comment]
Hi, thanks for contribution! I don't see any issues in this implementation, I may suggest only to skip this test and dig for root cause separately.

P.S. Please, rebase your change, I broken your test implementation by refactoring.

**Comment by p-wysocki** [comment]
Hi @sydarb, are you still working on this?

**Comment by mlukasze** [comment]
hey @sydarb 
let us know are you willing to finish this PR, please :)

File: src/frontends/onnx/frontend/src/op/mish.hpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+OPENVINO_SUPPRESS_DEPRECATED_START
+
+#include "ngraph/node.hpp"
+#include "onnx_import/core/node.hpp"
+
+namespace ngraph {
+namespace onnx_import {
+namespace op {
+namespace set_1 {
+OutputVector mish(const Node& node);
```
Comment (mitruska) [inline src/frontends/onnx/frontend/src/op/mish.hpp:17]:
```suggestion
OutputVector mish(const ov::frontend::onnx::Node& node);
```

File: src/frontends/onnx/frontend/src/op/mish.hpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// Copyright (C) 2018-2023 Intel Corporation
```
Comment (mitruska) [inline src/frontends/onnx/frontend/src/op/mish.hpp:1]:
```suggestion
// Copyright (C) 2018-2024 Intel Corporation
```

File: src/frontends/onnx/frontend/src/op/mish.cpp [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+#include <vector>
+
+#include "default_opset.hpp"
+OPENVINO_SUPPRESS_DEPRECATED_START
+
+namespace ngraph {
+namespace onnx_import {
+namespace op {
+namespace set_1 {
+OutputVector mish(const Node& node) {
```
Comment (mitruska) [inline src/frontends/onnx/frontend/src/op/mish.cpp:17]:
```suggestion
OutputVector mish(const ov::frontend::onnx::Node& node) {
```

File: src/frontends/onnx/frontend/src/op/mish.cpp [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+}  // namespace set_1
+
+}  // namespace op
+
+}  // namespace onnx_import
+
+}  // namespace ngraph
+OPENVINO_SUPPRESS_DEPRECATED_END
```
Comment (mitruska) [inline src/frontends/onnx/frontend/src/op/mish.cpp:29]:
```suggestion
}  // namespace set_1
}  // namespace op
}  // namespace onnx
}  // namespace frontend
}  // namespace ov
```

File: src/frontends/onnx/frontend/src/op/mish.cpp [hunk 5]
```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2018-2023 Intel Corporation
```
Comment (mitruska) [inline src/frontends/onnx/frontend/src/op/mish.cpp:1]:
```suggestion
// Copyright (C) 2018-2024 Intel Corporation
```

File: src/bindings/python/tests_compatibility/__init__.py
Comment (mitruska) [inline src/bindings/python/tests_compatibility/__init__.py:1]: This change is not needed, compatibility API has been removed, this file no longer exists.

File: src/bindings/python/tests_compatibility/test_onnx/test_backend.py
Comment (mitruska) [inline src/bindings/python/tests_compatibility/test_onnx/test_backend.py:1]: This change is not needed, compatibility API has been removed, this file no longer exists.

**Comment by mitruska** [comment]
As no response from the author, the task has been reassigned and implemented within another PR:
- https://github.com/openvinotoolkit/openvino/pull/22833

**Pull request closed without merging**